test/pretty_print_test.exs

```diff
@@ -342,7 +342,7 @@ defmodule Erlex.Test.PretyPrintTest do
     pretty_printed = Erlex.pretty_print(input)
 
     expected_output =
-      "(:Atom, :Encoding) :: binary() when Atom: atom(), Encoding: :latin1 | :utf8"
+      "(Atom, Encoding) :: binary() when Atom: atom(), Encoding: :latin1 | :utf8"
```

Comment:
Is this legal Elixir?

Reply:
This is my only major issue with the PR, other than the general hackiness of it. I think I can live with the hackiness if a typespec with this output is in fact legal. We can make it cleaner later, if this code actually works in a spec.

Reply:
Yeah, I wondered about that, but not overly long. That string was the Erlangified version - hence the capital letter var names. I found the issue it came from:

https://github.com/jeremyjh/dialyxir/issues/234

Reply:
If you don't mind verifying that this output on a `MyAtom.to_string` function with the pretty printed typespec produced here actually works in Dialyzer in a demo project, I will merge this PR.

Reply:
Yes it looks just like that:

```
The function call will not succeed.

:erlang.atom_to_binary("not an atom", :utf8)

breaks the contract
(Atom, Encoding) :: binary() when Atom: atom(), Encoding: :latin1 | :unicode | :utf8
```

Reply:
Into it, then.

Reply:
I may have misunderstood. If you mean, can I create a typespec that looks like

```elixir
@spec (Atom, Encoding) :: binary() when Atom: atom(), Encoding: :latin1 | :unicode | :utf8
``` 
in Elixir, the answer is no - neither the old version or new version will compile. I'm pretty sure you can only have a typespec look like that for code written in Erlang, [where this is coming from](https://github.com/erlang/otp/blob/master/erts/preloaded/src/erlang.erl#L345).

Reply:
Really it can be fixed, I think in addition to trimming ":" we could lowercase any name used in a when, which would produce a valid Elixir typespec here.

Reply:
@asummers sorry for failing at Github. I originally missed your second comment and thought your third was asking if I could reproduce #jeremyjh/dialyxir#234 and see that message print. 

It only dawned on me after you merged this that you want to be sure that what we pretty print will compile in Elixir. This should be obvious to me already :). But the original never compiled.

 I think we'll have this problem every single time we print a contract with `when` that comes from Erlang since Erlang variable names, including typevars have to be upper-case. They are not atoms though! They are just variable names, so we should just lowercase whatever comes from a `when` is my $.02.

Reply:
Hmmm. Can you open a PR with that change and we can discuss there? The beauty of low semver numbers =) I worry about an atom of the form `:foo_Bar` in the naive approach there.